src/details/ArborX_DetailsUtils.hpp

```diff
@@ -526,9 +526,21 @@ void reallocWithoutInitializing(View &v,
                                 size_t n6 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                                 size_t n7 = KOKKOS_IMPL_CTOR_DEFAULT_ARG)
 {
-  static_assert(View::is_managed, "Can only realloc managed views");
-  v = View(Kokkos::view_alloc(Kokkos::WithoutInitializing, v.label()), n0, n1,
-           n2, n3, n4, n5, n6, n7);
+  size_t ranks[8] = {n0, n1, n2, n3, n4, n5, n6, n7};
+  bool allocation_necessary = false;
+  for (unsigned int dim = 0; dim < v.rank_dynamic; ++dim)
+    if (ranks[dim] != v.extent(dim))
```

Comment:
There's no easy way to get all the extents (dynamic) in a single sweep, is there? I'd really like to have `if (ranks != extents) {}`.

Reply:
I am not aware of a member function to the `View` class that returns all `extents` in a container.

Reply:
Actually why is it not `rank`?

Reply:
I think so
```C++
Kokkos::View<int****[1][2][3][4]> bla("bla", 5,6,7,8);
std::cout << "rank: " << bla.rank << std::endl;
std::cout << "rank_dynamic: " << bla.rank_dynamic << std::endl;
std::cout << "extent: ";
for(unsigned int i=0; i<8; ++i)
   std::cout << bla.extent(i) << ' ';
std::cout << std::endl;
std::cout << "static_extent: ";
for(unsigned int i=0; i<8; ++i)
   std::cout << bla.static_extent(i) << ' ';
std::cout << std::endl;
```
gives
```
rank: 8
rank_dynamic: 4
extent: 5 6 7 8 1 2 3 4 
static_extent: 0 0 0 0 1 2 3 4 
```
In particular, we can only specify dynamic extents in the constructor.

Reply:
But you'd need to check all extents including static ones

Reply:
Aren't the static ones already part of the type?

Reply:
OK I see your point.  Fine by me.